app/controllers/schedule_controller.rb

```diff
@@ -9,13 +9,27 @@ def index
     end
   end
 
+  
   def search
-      course = params.fetch("search")
-      @search_result = Course.where({:course_number=> course}).first
-      redirect_to "/schedule"
+
+    course_number = params.fetch(:courseNum)
+    result = Course.where(course_number: course_number).first
```

Comment:
You can use [`find_by`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by) instead of `where` here since you're looking for one record.

```suggestion
    result = Course.find_by(course_number: course_number)
```